src/fits.cc

```diff
@@ -912,6 +936,131 @@ void Fits::writeImageImpl(T const *data, int nElements) {
     }
 }
 
+namespace {
+
+/// RAII for activating compression
+///
+/// Compression is a property of the file in cfitsio, so we need to set it,
+/// do our stuff and then disable it.
+struct ImageCompressionContext {
+  public:
+    ImageCompressionContext(Fits & fits_, ImageCompressionOptions const& useThis)
+        : fits(fits_) {
+        fits.setImageCompression(useThis);
+    }
+    ~ImageCompressionContext() {
+        int status = 0;
+        fits_unset_compression_request(reinterpret_cast<fitsfile *>(fits.fptr), &status);
```

Comment:
cfitsio has had bugs caused by not modifying its settings appropriately, and I thought this might be a way to guard against being caught out (again). But that's not what this context manager should do --- it should restore the previous settings. I've update this and moved `fits_unset_compression_request` into `Fits::setImageCompression`.